src/ln/channelmanager.rs

```diff
@@ -5119,6 +5135,63 @@ mod tests {
 		assert!(nodes[2].node.list_channels().is_empty());
 	}
 
+	#[test]
+	#[should_panic]
+	fn bolt2_open_channel_sending_node_checks1() { //This test needs to be on its own as we are catching a panic
+		let nodes = create_network(2);
+
+		// BOLT #2 spec: Sending node must ensure temporary_channel_id is unique from any other channel ID with the same peer.
+		// Note: currently a collision on a channel id will panic, not creating the channel with an error will be better as the system can still continue to function
```

Comment:
We expect the rng to work, so panic!()ing if the rng gives us the same 32 bytes that we saw somewhere is is *absolutely* a cause for panic/shutdown/burn it to the ground. Much, much better to panic in that case and make the user reconsider how broken their hardware/something is than continue in a very unsafe environment.

Reply:
Note was removed.